engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/distribute/DeploymentRedistributor.java

```diff
@@ -28,15 +30,15 @@ public DeploymentRedistributor(
 
   @Override
   public void onRecovered(final ReadonlyStreamProcessorContext context) {
-    final var writers = context.getWriters();
-
     final var deploymentDistributionBehavior =
         new DeploymentDistributionBehavior(
-            writers, partitionsCount, deploymentDistributionCommandSender);
+            context.getWriters(), partitionsCount, deploymentDistributionCommandSender);
 
-    deploymentState.foreachPendingDeploymentDistribution(
-        (key, partitionId, deployment) ->
-            deploymentDistributionBehavior.distributeDeploymentToPartition(
-                partitionId, key, deployment));
+    final var schedulingService = context.getScheduleService();
+    schedulingService.runAtFixedRate(
+        DEPLOYMENT_REDISTRIBUTION_INTERVAL,
+        () ->
+            deploymentState.foreachPendingDeploymentDistribution(
+                deploymentDistributionBehavior::distributeDeploymentToPartition));
```

Comment:
💭 I wonder if it's worth keeping the pending deployments cached in memory, filling it once on startup, instead of having to query every 10 seconds.

Reply:
IMO this is fine - usually there are 0 pending deployments so querying should be fast and in-memory caching adds a risk that we don't invalidate/populate the cache properly while adding little value. Let's let RocksDB do the caching for us :)

Reply:
Although I do wonder if we shouldn't increase the interval from 10 seconds to something like 30. WDYT?

Reply:
What was the interval before? I would keep it the same

Reply:
It was retried when the network request wasn't answered within 15 seconds. That's a bit different to the current behavior though, because now we actually need to process the record on the receiving partition before even attempting to acknowledge the distribution. I think that's a good argument to increase this slightly to 30 seconds as buffer for partitions that are processing slowly.

Now that I think about it: We should really use a backoff here.

Reply:
I did consider adding something like resilience4j or exposing methods in `SchedulingService` that make use of our `RetryStrategy`s but I thought that a simple exponential backoff is simple enough to just implement ad-hoc. Let me know if you think otherwise :bow: